database.psql

```diff
@@ -0,0 +1,696 @@
+FROM group_resource_privilege p 
+    LEFT JOIN resources r on p.resource_id=r.resource_id;
+
+-- OLD -- Eliminated 2015-03-28 in favor of marking public
+-- OLD -- resource as explicitly "held". 
+-- OLD ---------------------------------------------------
+-- OLD -- union of immutability filter and public resources 
+-- OLD ---------------------------------------------------
+-- OLD CREATE VIEW total_group_resource_privilege AS
+-- OLD   (select * from filtered_group_resource_privilege)
+-- OLD UNION
+-- OLD   (select * from public_group_resource_privilege) ;
+-- OLD 
+-- OLD ---------------------------------------------------
+-- OLD -- duplicate elimination and privilege fusion
+-- OLD ---------------------------------------------------
+-- OLD CREATE VIEW cumulative_group_resource_privilege AS
+-- OLD SELECT group_id, resource_id, MIN(privilege_id) as privilege_id
+-- OLD FROM total_group_resource_privilege 
+-- OLD GROUP BY group_id, resource_id; 
+
+---------------------------------------------------
+-- DEBUGGING views depict things in human-readable form
+---------------------------------------------------
+CREATE VIEW debug_membership_in_group AS
+SELECT u.user_login, g.group_name, p.privilege_code 
+from user_membership_in_group m left join users u on u.user_id=m.user_id 
+left join groups g on g.group_id=m.group_id 
+left join privileges p on p.privilege_id = m.privilege_id;
+
+CREATE VIEW debug_cumulative_user_resource_privilege AS 
+SELECT u.user_login, 
+    r.resource_title, r.resource_public, r.resource_immutable, 
+    p.privilege_code
+from cumulative_user_resource_privilege as m
+left join users u on u.user_id=m.user_id 
+left join resources r on r.resource_id=m.resource_id
+left join privileges p on p.privilege_id = m.privilege_id;
+
+CREATE VIEW debug_cumulative_group_resource_privilege AS 
+SELECT g.group_name, 
+    r.resource_title, r.resource_public, r.resource_immutable, 
+    p.privilege_code
+from cumulative_group_resource_privilege as m
+left join groups g on g.group_id=m.group_id 
+left join resources r on r.resource_id=m.resource_id
+left join privileges p on p.privilege_id = m.privilege_id;
+
+CREATE VIEW debug_group_access_to_resource AS
+SELECT g.group_name, 
+    r.resource_title, r.resource_public, r.resource_immutable, 
+    p.privilege_code, u2.user_login as asserting_login
+from group_access_to_resource as m
+left join groups g on g.group_id=m.group_id 
+left join resources r on r.resource_id=m.resource_id
+left join privileges p on p.privilege_id = m.privilege_id
+left join users u2 on u2.user_id=m.assertion_user_id;
+
+CREATE VIEW debug_user_access_to_group AS
+SELECT u.user_login, 
+    g.group_name,  
+    p.privilege_code, 
+    u2.user_login as asserting_login
+from user_access_to_group as m
+left join users u on u.user_id=m.user_id 
+left join groups g on g.group_id=m.group_id
+left join privileges p on p.privilege_id = m.privilege_id
+left join users u2 on u2.user_id=m.assertion_user_id;
+
```

Comment:
@ironfroggy Yes, this is used as initial schema to initialize HSAccess database. I am currently using the same file name as used by Alva in his branch, but have intended to rename it to pg.hs_access_init_db.sql to following the same naming pattern for Django-managed HydroShare database initialization script (i.e., pg.hs_init_db.sql). This HSAccess database only reside together with Django Hydroshare database for release 1.5 as an intermediate step, and will be moved to iRODS side sitting together with iCAT (iRODS catalog database) for release 1.6.